README.md

```diff
@@ -1,109 +1,233 @@
 ![build](https://github.com/nhosoya/omniauth-apple/workflows/RSpec/badge.svg?branch=master&event=push)
 
-# OmniAuth::Apple
+# OmniAuth Apple Strategy
 
 OmniAuth strategy for [Sign In with Apple](https://developer.apple.com/sign-in-with-apple/).
 
 ## Installation
 
 Add this line to your application's Gemfile:
+```ruby
+gem 'omniauth-apple', '~> 1.3.0'
+```
+
+Then execute
+```bash
+bundle install
+```
+
+Or install it yourself globally with
+```bash
+gem install omniauth-apple
+```
+
+## Usage
+Using Devise ? Skip to <a href="#use-with-devise">Use with Devise</a>
 
+Here's an example for adding the middleware to a Rails app in `config/initializers/omniauth.rb`
 ```ruby
-gem 'omniauth-apple'
+Rails.application.config.middleware.use OmniAuth::Builder do
+  provider :apple, ENV['APPLE_CLIENT_ID'], '', {
+    key_id: ENV['APPLE_KEY_ID'],
+    pem: ENV['APPLE_PRIVATE_KEY'],
+    scope: 'email name',
+    team_id: ENV['APPLE_TEAM_ID']
+  }
+end
 ```
 
-And then execute:
+You can find more confiduration options in the <a href="#configuration">Configuration</a> section.
 
-    $ bundle
+NOTE: Any change made to the middleware's configuration will required you to reset your server before taking action.
 
-Or install it yourself as:
+### Use with Devise
+When using `omniauth-apple` with Devise you must omit the `config/initializers/omniauth.rb` file.
+Instead you can add the middleware's configuration in `config/initializers/devise.rb`
+```ruby
+# config/initializers/devise.rb
+config.omniauth :apple, 'APPLE_CLIENT_ID', '', {
+  key_id: ENV['APPLE_KEY_ID'],
+  pem: ENV['APPLE_PRIVATE_KEY'],
+  scope: 'email name',
+  team_id: ENV['APPLE_TEAM_ID']
+}
+```
 
-    $ gem install omniauth-apple
+Make sure to your Omniauthable model includes the new provider. Generally that model is your `User` model.
 
-## Usage
+You should also create a class method to register and find users from a omniauth provider's callback in your model.
+```ruby
+# app/models.user.rb
+def User < ApplicationRecord
+  devise :omniauthable, omniauth_providers: %i[apple]
+
+  class << self
+    def from_omniauth(access_token)
+      user = find_or_initialize_by(email: access_token.info['email'])
+      user.update!(name: access_token.info['name'], password: Devise.friendly_token[0, 20]) unless user.persisted?
+      user
+    end
+  end
```

Comment:
This is slightly more Rails style:

```suggestion
  def self.from_omniauth(access_token)
    user = find_or_create_by(email: access_token.info['email']) do |new_user|
      new_user.name = access_token.info['name']
      new_user.password = Devise.friendly_token 20
    end
    user
  end
```

but I don't think that's exactly right either, because `info` is only returned on first sign in.

Reply:
> This is slightly more Rails style

I've seen both equally, I don't think there's a convention about which to use (at least not in https://github.com/rubocop/rails-style-guide) and I prefer `class << self` because it allows creating private class methods

> I don't think that's exactly right either, because info is only returned on first sign in.

I tried logging in multiple times in a row with the same apple account and I can confirm the `request.env['omniauth.auth'][:info]` is present each time so I'm confident it is right as is